swift_package/BrainFlowCI/BrainFlowBindings/BoardDescription.swift

```diff
@@ -0,0 +1,59 @@
+import Foundation
+
+/// BoardDescription formalizes the attributes of a BrainFlow board via a structured JSON object.
+struct BoardDescription: Codable, Equatable {
+    var package_num_channel: Int32 = 0
+    var timestamp_channel: Int32 = 0
+    var accel_channels: [Int32] = [Int32]()
+    var ecg_channels: [Int32] = [Int32]()
+    var eeg_channels: [Int32] = [Int32]()
+    var eeg_names: String = ""
+    var emg_channels: [Int32] = [Int32]()
+    var eog_channels: [Int32] = [Int32]()
+    var marker_channel: Int32 = 0
+    var name: String = ""
+    var num_rows: Int32 = 0
+    var sampling_rate: Int32 = 0
+    var battery_channel: Int32 = 0
+    var eda_channels: [Int32] = [Int32]()
+    var gyro_channels: [Int32] = [Int32]()
+    var ppg_channels: [Int32] = [Int32]()
+    var resistance_channels: [Int32] = [Int32]()
+    var temperature_channels: [Int32] = [Int32]()
+        
+    // decode the input JSON into self:
+    init(_ descriptionJSON: String?) throws {
```

Comment:
if you've added a special method fromJSON, why do we still have regulat constructor from json string?

Reply:
Not sure. I will take a look at removing the constructor.